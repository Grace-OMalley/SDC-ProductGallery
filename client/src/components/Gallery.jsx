
import React from 'react';
import ImageRow from './ImageRow.jsx';
class Gallery extends React.Component {

  constructor (props) {
    super(props);
    this.state = {
      selected: 0
    };
  }

  mouseHandler(index) {
    this.setState({selected: +index.currentTarget.dataset.index});
  }

  render() {
    console.log(this.state);
    return (
      <div className="imageBlock">
        <img src={this.props.images[this.state.selected]} />
        <div className="imageCaption">
          Click image to open expanded view
        </div>
        <ImageRow images={this.props.images} mouseHandler={(e) => this.mouseHandler(e)} selected={this.state.selected}/>
      </div>
    );
  }
}

export default Gallery;