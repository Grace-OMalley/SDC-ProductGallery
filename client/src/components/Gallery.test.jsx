import React from 'react';
import renderer from 'react-test-renderer';
import {cleanup, waitFor, waitForElementToBeRemoved, fireEvent, render, queryHelpers, buildQueries, screen} from '@testing-library/react';
import Gallery from './Gallery';

it('renders correctly', () => {
  const mock = jest.fn();
  const tree = renderer
    .create(<Gallery
      images={[1, 2, 3]}
      openHandler={mock}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('passes click fn to image', () => {
  const mock = jest.fn();
  const {container} = render(
    <Gallery
      images={[1, 2, 3]}
      openHandler={mock}
    />
  );
  const img = container.querySelector('img');
  fireEvent.click(img);
  expect(mock).toHaveBeenCalledTimes(1);
});

it('changes main image to match hovered-over image', () => {
  const {container} = render(
    <Gallery
      images={[1, 2, 3]}
    />
  );
  const img = container.querySelector('img');
  expect(img.getAttribute('src')).toBe('1');

  const thumb = container.querySelector('ul').children[1];
  fireEvent.mouseEnter(thumb);
  expect(img.getAttribute('src')).toBe('2');
});

it('does nothing when thumbnails are clicked', () => {
  const {container} = render(
    <Gallery
      images={[1, 2, 3]}
    />
  );
  const img = container.querySelector('img');
  expect(img.getAttribute('src')).toBe('1');
  const thumb = container.querySelector('ul').children[1];
  fireEvent.click(thumb);
  expect(container).toMatchSnapshot();
  expect(img.getAttribute('src')).toBe('1');
});