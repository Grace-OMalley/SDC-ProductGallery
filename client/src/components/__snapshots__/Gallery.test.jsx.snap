// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does nothing when thumbnails are clicked 1`] = `
<div>
  <div
    class="imageBlock"
  >
    <img
      src="1"
    />
    <div
      class="imageCaption"
    >
      Click image to open expanded view
    </div>
    <ul
      class="imageRow"
    >
      <li
        class="selected"
        data-index="0"
      >
        <img
          src="1"
        />
      </li>
      <li
        class=""
        data-index="1"
      >
        <img
          src="2"
        />
      </li>
      <li
        class=""
        data-index="2"
      >
        <img
          src="3"
        />
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="imageBlock"
>
  <img
    onClick={[MockFunction]}
    src={1}
  />
  <div
    className="imageCaption"
  >
    Click image to open expanded view
  </div>
  <ul
    className="imageRow"
  >
    <li
      className="selected"
      data-index={0}
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <img
        src={1}
      />
    </li>
    <li
      className=""
      data-index={1}
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <img
        src={2}
      />
    </li>
    <li
      className=""
      data-index={2}
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <img
        src={3}
      />
    </li>
  </ul>
</div>
`;
